app/components/common/form/ImageDropzoneField.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { useCallback, useState } from 'react';
+import Dropzone from 'react-dropzone';
+import { Controller } from 'react-hook-form';
+
+import CloudUploadIcon from '@mui/icons-material/CloudUpload';
+import Box from '@mui/material/Box';
+import FormHelperText from '@mui/material/FormHelperText';
+import Typography from '@mui/material/Typography';
+
+import { FormInputProps } from '@/common/formTypes';
+import { invalid_file_size } from '@/common/formValidation';
+import { clientConfig } from '@/config/client';
+import * as m from '@/src/paraglide/messages.js';
+import theme from '@/styles/theme';
+
+import ImagePreview from './ImagePreview';
+
+interface ImageDropzoneFieldProps extends FormInputProps {
+  setValue: (name: any, value: File | null, options?: { shouldDirty: boolean }) => void;
+}
+
+export const ImageDropzoneField = ({ name, control, setValue }: ImageDropzoneFieldProps) => {
+  const [file, setFile] = useState<string | null>();
+
+  const onDrop = useCallback(
+    (acceptedFiles: File[]) => {
+      if (acceptedFiles.length > 0) {
+        const file = acceptedFiles[0];
+        setFile(URL.createObjectURL(file));
+        setValue(name, file, { shouldDirty: true });
+      }
+    },
+
+    [name, setValue],
+  );
+
+  const onCancel = () => {
+    setFile(null);
+    setValue(name, null, { shouldDirty: true });
+  };
+
+  const fileValidator = (file: File) => {
+    if (file.size > clientConfig.NEXT_PUBLIC_BODY_SIZE_LIMIT) {
+      return {
+        code: 'invalid-size',
+        message: invalid_file_size.message,
+      };
+    }
+    return null;
+  };
+
+  return (
+    <Controller
+      control={control}
+      name={name}
+      render={({ field: { onBlur, value } }) => (
+        <>
+          <Dropzone
+            onDrop={async (acceptedFiles: File[]) => {
+              onDrop(acceptedFiles);
+            }}
+            validator={fileValidator}
+            accept={{ 'image/*': [] }}
+            multiple={false}
+          >
+            {({ getRootProps, getInputProps, isFocused, isDragReject, fileRejections }) => {
+              const style = {
+                ...baseStyle,
+                ...(isFocused ? focusedStyle : {}),
+                ...(isDragReject ? rejectedStyle : {}),
+              };
+
+              const fileRejectionItems =
+                fileRejections &&
+                fileRejections[0].errors.map((e) => (
+                  <FormHelperText key={e.code} sx={{ color: 'formText.main' }}>
+                    {e.message}
+                  </FormHelperText>
+                ));
```

Comment:
```suggestion
              const fileRejectionItems = fileRejections?.[0]?.errors.map((e) => (
                <FormHelperText key={e.code} sx={{ color: 'formText.main' }}>
                  {e.message}
                </FormHelperText>
              ));
```
I get an error when I access the user page because fileRejections is an empty list so the entry [0] is undefined

Reply:
Thanks for noticing. Fixed.